consensus/src/types.rs

```diff
@@ -1,73 +1,417 @@
 //! Consensus types shared across the crate.
+//!
+//! This module defines the core types used throughout the consensus implementation:
+//!
+//! - [`Epoch`]: Represents a distinct set of validators. When the validator set changes,
+//!   the epoch increments. Epochs provide reconfiguration boundaries for the consensus protocol.
+//!
+//! - [`View`]: A monotonically increasing counter within a single epoch, representing individual
+//!   consensus rounds. Views advance as the protocol progresses through proposals and votes.
+//!
+//! - [`Round`]: Combines an epoch and view into a single identifier for a consensus round.
+//!   Provides ordering across epoch boundaries.
+//!
+//! - [`Delta`]: A generic type representing offsets or durations for consensus types. Provides
+//!   type safety to prevent mixing epoch and view deltas. Type aliases [`EpochDelta`] and
+//!   [`ViewDelta`] are provided for convenience.
+//!
+//! # Arithmetic Safety
+//!
+//! Arithmetic operations avoid silent errors. Only `next()` panics on overflow. All other
+//! operations either saturate or return `Option`.
+//!
+//! # Type Conversions
+//!
+//! Explicit type constructors (`Epoch::new()`, `View::new()`) are required to create instances
+//! from raw integers. Implicit conversions via, e.g. `From<u64>` are intentionally not provided
+//! to prevent accidental type misuse.
+//!
+//! For convenience, `From` implementations are enabled under `#[cfg(test)]`.
 
 use bytes::{Buf, BufMut};
-use commonware_codec::{EncodeSize, Error, Read, ReadExt, Write};
-use std::fmt::Display;
+use commonware_codec::{varint::UInt, EncodeSize, Error, Read, ReadExt, Write};
+use std::{
+    fmt::{self, Display, Formatter},
+    marker::PhantomData,
+};
 
-/// Epoch is the type used to represent a distinct set of validators.
+/// Represents a distinct set of validators in the consensus protocol for a contiguous
```

Comment:
nit: `Represents a distinct segment of a contiguous sequence of views.`

(we assign validators as a result but I don't think the raw epoch type needs to know that)